trestle/utils/fs.py

```diff
@@ -61,6 +62,47 @@ def get_trestle_project_root(path: pathlib.Path) -> Optional[pathlib.Path]:
     return None
 
 
+def verify_trestle_folder(path: pathlib.Path) -> bool:
+    """Trestle folder should not have any files other than readme and models."""
+    is_valid = True
+    for file_path in path.rglob('*'):
+        if file_path.is_file() and not local_and_visible(file_path):
+            logger.warning(f'Hidden files and symlinks are not allowed in OSCAL directories, deleting: {file_path}.')
+            os.remove(file_path)
+        if file_path.is_file() and file_path.suffix not in {'.json', '.xml', '.yaml', '.yml', '.md'}:
+            logger.warning(
+                f'Files of {file_path.suffix} are not allowed in the OSCAL directories '
+                f'and can cause the issues. Please remove the file {file_path}'
+            )
+            is_valid = False
+
+    return is_valid
+
+
+def check_oscal_directories(root_path: pathlib.Path) -> bool:
+    """
+    Identify the state of the Trestle workspace.
+
+    Traverses Trestle workspace and looks for unexpected files or directories.
+    Additional files are allowed in the Trestle root but not inside the model folders.
+    """
+    trestle_dir_walk = os.walk(root_path)
+    is_valid = True
+
+    for _, dirs, _ in trestle_dir_walk:
+        for d in dirs:
+            if d in MODEL_DIR_LIST:
+                is_valid = verify_trestle_folder(root_path / d)
+                if not is_valid:
+                    logger.error(
+                        f'OSCAL directory {root_path / d} contains unsupported files, '
```

Comment:
Does this work if someone has a *file* named catalogs in the main dir?

Reply:
As per discussion we allow any files and folders in the main directory, thus I only check the contents of OSCAL directories

Reply:
This would definitely be a problem if there is a *file* named catalogs instead of a *directory* named catalogs.  I don't think we ever discussed this particular oddball case.  I'm just making sure that if allowed directories are present: catalogs, profiles, etc. - they really are directories and not files.  I wasn't clear what your code would do there.

Reply:
I only iterate over directories in the main folder, I don't consider the files. We can always restrict users to not have files with such names as well